vscode/src/ghCopilot.ts

```diff
@@ -0,0 +1,226 @@
+      ResetAll(qs);
+      return results;
+  }
+  \`\`\`
+
+  The \`qsharp_widgets\` package provides viewers for circuits and histograms, e.g.
+
+  \`\`\`python
+  from qsharp_widgets import Circuit, Histogram
+  Circuit(qsharp.circuit("GHZSample(3)"))
+  \`\`\`
+
+  Note that the latest Q# and QDK releases don't require or use the old IQ# kernel. It just needs to the \`qsharp\` PyPI package,
+  and maybe \`qsharp_widgets\` for visuals.`;
+
+/**
+ * Checks if the copilot-instructions.md file exists and contains Q# instructions.
+ *
+ * @param workspaceFolder The workspace folder URI.
+ * @returns A promise that resolves to true if the file exists and contains Q# instructions.
+ */
+async function hasQSharpCopilotInstructions(
+  workspaceFolder: vscode.Uri,
+): Promise<boolean> {
+  try {
+    const githubDir = vscode.Uri.joinPath(workspaceFolder, ".github");
+    const instructionsFile = vscode.Uri.joinPath(
+      githubDir,
+      "copilot-instructions.md",
+    );
+
+    // Check if file exists
+    await vscode.workspace.fs.stat(instructionsFile);
+
+    // Check if file contains Q# instructions
+    const existingContent =
+      await vscode.workspace.fs.readFile(instructionsFile);
+    const existingText = new TextDecoder("utf-8").decode(existingContent);
+
+    return existingText.includes(codingInstructionsTitle);
+  } catch {
+    // If any error occurs (file not found, etc.), return false
+    return false;
+  }
+}
+
+/**
+ * Command to update or create the Copilot instructions file for Q#.
+ * Shows a prompt to the user and updates the file if confirmed.
+ */
+async function updateGhCopilotInstructionsCommand() {
+  const workspaceFolders = vscode.workspace.workspaceFolders;
+  if (!workspaceFolders || workspaceFolders.length === 0) {
+    vscode.window.showErrorMessage("No workspace folder is open");
+    return;
+  }
+
+  for (const folder of workspaceFolders) {
+    // Check if the file already exists with Q# instructions
+    const hasInstructions = await hasQSharpCopilotInstructions(folder.uri);
+    if (hasInstructions) {
+      vscode.window.showInformationMessage(
+        "copilot-instructions.md already contains Q# instructions",
+      );
+      return;
+    }
+  }
+
+  const workspaceFolder = workspaceFolders[0].uri;
+
+  return await updateCopilotInstructions(workspaceFolder);
+}
+
+async function updateCopilotInstructions(workspaceFolder: vscode.Uri) {
+  // Show a yes/no prompt to the user
+  const response = await vscode.window.showInformationMessage(
+    "We're about to update your `copilot-instructions.md` file.\n\n" +
+      "This file helps GitHub Copilot understand and work better with Q# files and features provided by the Quantum Development Kit extension.\n\n" +
+      "Would you like to proceed with updating `copilot-instructions.md`?",
+    { modal: true },
+    "Yes",
+    "No",
+  );
+
+  if (response !== "Yes") {
+    return; // User canceled or dismissed the dialog
```

Comment:
Can be a follow up, but might be an interesting telemetry point to see how many folks bail out here.

Reply:
I went ahead and added the data point